File: example-webapp/src/main/java/vip/yazilim/libs/springvip/template/example/PersonServiceImpl.java
package vip.yazilim.libs.springvip.template.example;

import org.jetbrains.annotations.NotNull;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Service;
import vip.yazilim.libs.springvip.util.generic.service.impl.AGenericServiceCrud;

@Service
public class PersonServiceImpl extends AGenericServiceCrud<Person, Integer> implements IPersonService {

    public PersonServiceImpl(@NotNull JpaRepository<Person, Integer> repository) {
        super(repository, Person.class, Integer.class);
    }

    @NotNull
    @Override
    protected Integer getId(@NotNull Person entity) {
        return entity.getId();
    }
}

File: src/main/kotlin/vip/yazilim/springvip/templates/kotlin/service/impl/PersonServiceImpl.kt
package vip.yazilim.springvip.templates.kotlin.service.impl

import org.springframework.stereotype.Service
import vip.yazilim.libs.springvip.util.generic.service.impl.AGenericServiceCrud
import vip.yazilim.springvip.templates.kotlin.entity.Person
import vip.yazilim.springvip.templates.kotlin.repo.IPersonRepo
import vip.yazilim.springvip.templates.kotlin.service.IPersonService

/**
 *
 * @author maemresen - maemresen@yazilim.vip
 * 12/30/2020
 */
@Service
class PersonServiceImpl(
    private val personRepository: IPersonRepo
) : AGenericServiceCrud<Person, Long>(personRepository, Person::class, Long::class), IPersonService {
    override fun getId(entity: Person): Long {
        return entity.id
    }
}